package/src/values/hooks/useDerivedValue.ts

```diff
@@ -12,9 +12,10 @@ type CreateDerivedvalue = typeof ValueApi.createDerivedValue;
  * @param values Depenedant values
  * @returns A readonly value
  */
-export const useDerivedValue: CreateDerivedvalue = <R>(
-  cb: (...args: Array<unknown>) => R,
-  values: Array<SkiaReadonlyValue<unknown>>
+export const useDerivedValue: CreateDerivedvalue = <R, A extends Array<unknown> = Array<unknown>>(
+  cb: (...args: A extends readonly [...infer Args] ? [...Args] : A) => R,
+  values: Array<SkiaReadonlyValue<A extends readonly [...infer Args] ? Args: unknown>>
 ): SkiaReadonlyValue<R> => {
-  return useMemo(() => ValueApi.createDerivedValue(cb, values), [cb, values]);
+  return useMemo(() => ValueApi.createDerivedValue(cb, values), values);
```

Comment:
@nandorojo thanks for the PR we would like #265 to superseed it. However, I would like to understand why the cb has been removed from the list of dependencies.
`[cb, values]` seems wrong too. Should it be `[cb, ...values]`?

Reply:
I believe including cb in the dependencies array  would negate the memoization altogether. 

The point of the dependency array is to say, only recompute whenever one of these dependencies changes. The callback is written in render, meaning its reference technically changes every render. If we include it in the deps array, then I believe that we are removing the point of the array in the first place. It should only have the dependencies _of_ the function, not the function itself.

Reply:
nice catch thanks 🙌🏼 We might be making this mistake in other places 🤔

Reply:
nice catch thanks 🙌🏼 We might be making this mistake in other places 🤔